tol_lab_share/messages/consumed/base_validator.py

```diff
@@ -0,0 +1,60 @@
+from abc import ABC, abstractmethod
+import logging
+from uuid import UUID
+
+from tol_lab_share import error_codes
+from tol_lab_share.error_codes import ErrorCode
+
+from .message_field import MessageField
+
+LOGGER = logging.getLogger(__name__)
+
+
+class BaseValidator(ABC):
+    def __init__(self) -> None:
+        super().__init__()
+        self._errors: list[ErrorCode] = []
+
+    @abstractmethod
+    def validate(self) -> bool: ...
+
+    @property
+    def errors(self) -> list[ErrorCode]:
+        """Gets the list of errors that occurred during validation.
+
+        Returns:
+            list[ErrorCode]: A list of errors that occurred during validation.
+        """
+        return self._errors.copy()
```

Comment:
Is the `_errors` list copied because the client is not supposed to modify it?

Reply:
It's something I picked up as a habit from iOS development.  We don't have immutable lists in Python and when you provide access to mutable types via a property, you return a copy instead.  So in other words, yes to your theory.  I realise I could also wrap the list as a tuple for a similar effect, but I felt that's changing the type which feels like I'm making it seem like we maintain it as a tuple internally which we don't.

Reply:
@yoldas makes a good point. I didn't know that [`copy` returns a shallow copy](https://docs.python.org/3/library/copy.html). I think it would be better to update it to `deepcopy`.

Reply:
Yes, and to explain my choices here.  You should protect your internal state, but the state of the ErrorCode instances is not the responsibility of this class.  It was given those objects by another object and didn't create them itself.  When this happens, you cannot guarantee immutability, so trying to make them immutable here is pointless.  The thing that gave you the object could have retained a reference to the object and passed it to other classes as well.

Ideally, Python would prefer immutability as the default for objects, but that's not what we're working with here and Python has the philosophy that we should all do the right thing and not put lots of effort into trying to programatically defend against data manipulation.  Hence why there's no true private/protected/public system in place.  You're supposed to know that you shouldn't be calling methods if they start with an underscore, but nothing stops you from doing so.